main_test.go

```diff
@@ -205,9 +208,58 @@ func (t *MainTest) TestStringifyShouldReturnAllFlagsPassedInFlagStorageAsMarshal
 	}
 
 	actual, err := util.Stringify(flags)
-	AssertEq(nil, err)
+	assert.Equal(t.T(), nil, err)
+
+	expected := "{\"AppName\":\"\",\"Foreground\":false,\"ConfigFile\":\"\",\"MountOptions\":{\"1\":\"one\",\"2\":\"two\",\"3\":\"three\"},\"DirMode\":0,\"FileMode\":0,\"Uid\":0,\"Gid\":0,\"ImplicitDirs\":false,\"OnlyDir\":\"\",\"RenameDirLimit\":0,\"IgnoreInterrupts\":false,\"CustomEndpoint\":null,\"BillingProject\":\"\",\"KeyFile\":\"\",\"TokenUrl\":\"\",\"ReuseTokenFromUrl\":false,\"EgressBandwidthLimitBytesPerSecond\":0,\"OpRateLimitHz\":0,\"SequentialReadSizeMb\":10,\"AnonymousAccess\":false,\"MaxRetrySleep\":0,\"StatCacheCapacity\":0,\"StatCacheTTL\":0,\"TypeCacheTTL\":0,\"HttpClientTimeout\":0,\"MaxRetryDuration\":0,\"RetryMultiplier\":0,\"LocalFileCache\":false,\"TempDir\":\"\",\"ClientProtocol\":\"http4\",\"MaxConnsPerHost\":0,\"MaxIdleConnsPerHost\":0,\"EnableNonexistentTypeCache\":false,\"StackdriverExportInterval\":0,\"OtelCollectorAddress\":\"\",\"LogFile\":\"\",\"LogFormat\":\"\",\"ExperimentalEnableJsonRead\":false,\"DebugFuseErrors\":false,\"DebugFuse\":false,\"DebugFS\":false,\"DebugGCS\":false,\"DebugHTTP\":false,\"DebugInvariants\":false,\"DebugMutex\":false,\"MetadataPrefetchMode\":\"\"}"
+	assert.Equal(t.T(), expected, actual)
+}
+
+func (t *MainTest) TestCallListRecursiveOnPopulatedDirectory() {
+	// Set up a mini file-system to test on.
+	rootdir, err := os.MkdirTemp("/tmp", "TestCallRecursive-*")
+	assert.False(t.T(), len(rootdir) == 0 || err != nil, "failed to create temp-directory for test. err = %v", err)
```

Comment:
As per our coding guidelens, the tests should have three segments: Arrange, Act and Assert. Let's follow that and remove asserts that are not really the behavior that we want to test.

Reply:
If we don't check the error on the `os.MkdirTemp` or the other functions, which we may have not written, it doesn't exempt us from check their error.

I removed the assert here and got the following error as expected:
`Error return value of `os.CreateTemp` is not checked (errcheck)`.

So, I have to put it back to let the linter shut up about it.

Reply:
@kislaykishore ^

Reply:
Instead of an assert you can instead use t.Fatal to ensure that while an assert in the ACT block is not distracting, the program fails if an error is returned.

Reply:
SG